lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -55,13 +55,16 @@ export class RedirectHandler extends InteractionHandler {
         // Interaction is completed - remove interaction status.
         this.browserStorage.removeItem(this.browserStorage.generateCacheKey(BrowserConstants.INTERACTION_STATUS_KEY));
 
+        const hashUrlString = new UrlString(locationHash);
+        // Deserialize hash fragment response parameters.
+        const serverParams: ServerAuthorizationCodeResponse = UrlString.getDeserializedHash(hashUrlString.getHash());
```

Comment:
We should consider making a helper function to get the state from the hash. We're doing this in a couple places and I've seen a couple customer requests to provide a method to extract the state (scenarios such as subdomain routing would require them to handle their own redirects and they use state to do that)

Reply:
Yep, working on this now.

Reply:
Added.

Reply:
We already return the state in the response. We can make a helper for this but there shouldn't be any reason that they can't use the state to do the subdomain today.

Reply:
In this scenario they aren't going as far as getting the response because they need to do another redirect themselves before processing the hash. This doesn't need to be exposed as a public API in this PR but wanted to point it out as another use case for a helper. We can talk offline if you want to clarify the use case I'm referring to here